tests/foreman/api/test_repositories.py

```diff
@@ -186,6 +191,41 @@ def test_positive_sync_kickstart_repo(module_sca_manifest_org, target_sat):
     assert rh_repo.content_counts['rpm'] > 0
 
 
+def test_positive_sync_upstream_repo_with_zst_compression(
+    module_org, module_product, module_target_sat
+):
+    """Sync upstream repo having zst compression and verify it succeeds.
+
+    :id: 1eddff2a-b6b5-420b-a0e8-ba6a05c11ca4
+
+    :expectedresults: Repo sync is successful and no zst type compression errors are present in /var/log/messages.
+
+    :customerscenario: true
+
+    :steps:
+
+        1. Sync upstream repository having zst type compression.
+        2. Assert that no errors related to compression type  are present in
+            /var/log/messages.
+        3. Assert that sync was executed properly.
+
+    :CaseComponent: Pulp
+
+    :BZ: 2241934
+    """
+    repo = module_target_sat.api.Repository(
+        product=module_product, content_type='yum', url=f'{PULP_FIXTURE_ROOT}/rpm-zstd-metadata'
```

Comment:
Here {PULP_FIXTURE_ROOT}/rpm-zstd-metadata can be added as constant which can be leveraged in future
```suggestion
        product=module_product, content_type='yum', url=FAKE_ZST_REPO
```

Reply:
Updated